patcher/patcher.go

```diff
@@ -116,6 +116,51 @@ func PatchProc(pid uint32, moduleName string, offsetAddr uintptr, old []byte, ne
 
 	var offset = moduleInfo.LPBaseOfDll + offsetAddr
 
+	// Information about the contents of the memory to read
+	var memoryBasicInfo struct {
+		BaseAddress       uintptr
+		AllocationBase    uintptr
+		AllocationProtect uint32
+		PartitionId       uint32
+		RegionSize        uint64
+		State             uint32
+		Protect           uint32
+		Type              uint32
+	}
+
+	var memoryBasicInfoSize = uint32(unsafe.Sizeof(memoryBasicInfo))
+
+	var p uintptr = moduleInfo.LPBaseOfDll
+
+	// Programmatically find the offset of the API url
+	// Don't search beyond the end of the application memory
+	for p < moduleInfo.LPBaseOfDll+uintptr(moduleInfo.SizeOfImage) {
+		ret, _, err = procVirtualQueryEx.Call(uintptr(proc), p, uintptr(unsafe.Pointer(&memoryBasicInfo)), uintptr(memoryBasicInfoSize))
+		if ret == 0 {
+			return 0, fmt.Errorf("error in VirtualQueryEx: %w", err)
+		} else if ret == unsafe.Sizeof(memoryBasicInfo) {
+			var bytesRead uint32
+			var chunk = make([]byte, memoryBasicInfo.RegionSize)
```

Comment:
Probably ok in this case since the GGST binary is about 50-60MB, but pretty risky in general as regions can be up to `SIZE_T` in size.